client/my-sites/post-share/index.jsx

```diff
@@ -140,38 +278,55 @@ class PostShare extends Component {
 			'has-connections': this.hasConnections()
 		} );
 
+		const {
+			connections,
+			failure,
+			postId,
+			requesting,
+			siteId,
+			siteSlug,
+			success,
+			translate,
+		} = this.props;
+
+		if ( ! siteId || ! postId ) {
+			return null;
+		}
+
 		return (
 			<div className="post-share">
-				{ this.props.requesting &&
+				{ requesting &&
 					<Notice status="is-warning" showDismiss={ false }>
-							{ this.props.translate( 'Sharing…' ) }
+							{ translate( 'Sharing…' ) }
 					</Notice>
 				}
 
-				{ this.props.success &&
+				{ success &&
 					<Notice status="is-success" onDismissClick={ this.dismiss }>
-						{ this.props.translate( 'Post shared. Please check your social media accounts.' ) }
+						{ translate( 'Post shared. Please check your social media accounts.' ) }
 					</Notice>
 				}
 
-				{ this.props.failure &&
+				{ failure &&
 					<Notice status="is-error" onDismissClick={ this.dismiss }>
-						{ this.props.translate( 'Something went wrong. Please don\'t be mad.' ) }
+						{ translate( 'Something went wrong. Please don\'t be mad.' ) }
 						</Notice>
 				}
 
 				<div className={ classes }>
-					{ this.props.siteId && <QueryPostTypes siteId={ this.props.siteId } /> }
+					<QueryPublicizeConnections siteId={ siteId } />
+					<QueryPostTypes siteId={ siteId } />
+
 					<div className="post-share__head">
 						<h4 className="post-share__title">
-							{ this.props.translate( 'Share post' ) }
+							{ translate( 'Share this post' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **7** times:
`translate( 'Share this post!' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).